packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -476,17 +474,17 @@ class _BottomNavigationTile extends StatelessWidget {
               iconSize: iconSize,
               selected: selected,
               item: item,
-              selectedIconTheme: selectedIconTheme ?? bottomTheme.selectedIconTheme,
-              unselectedIconTheme: unselectedIconTheme ?? bottomTheme.unselectedIconTheme,
+              selectedIconTheme: selectedIconTheme,
+              unselectedIconTheme: unselectedIconTheme,
             ),
             _Label(
               colorTween: colorTween,
               animation: animation,
               item: item,
-              selectedLabelStyle: selectedLabelStyle ?? bottomTheme.selectedLabelStyle,
-              unselectedLabelStyle: unselectedLabelStyle ?? bottomTheme.unselectedLabelStyle,
-              showSelectedLabels: showSelectedLabels ?? bottomTheme.showUnselectedLabels,
-              showUnselectedLabels: showUnselectedLabels ?? bottomTheme.showUnselectedLabels,
+              selectedLabelStyle: selectedLabelStyle,
+              unselectedLabelStyle: unselectedLabelStyle,
+              showSelectedLabels: showSelectedLabels,
+              showUnselectedLabels: showUnselectedLabels,
```

Comment:
whats the reason for this change?

Reply:
The null coalescing here is redundant, because the values passed to `_BottomNavigationTile` have already done the null theme checks. This will also make the NNBD migration more clear.

https://github.com/flutter/flutter/pull/67078